framework/validators/UniqueValidator.php

```diff
@@ -174,6 +174,9 @@ private function modelExists($targetClass, $conditions, $model)
                 // only select primary key to optimize query
                 $columnsCondition = array_flip($targetClass::primaryKey());
                 $query->select(array_flip($this->applyTableAlias($query, $columnsCondition)));
+                
+                // since we only select() the primary key, a potential default scope can't be loaded
```

Comment:
this comment is still quite magical to me. Can it be made a bit more dummy proof?